### DISCUSSION TITLE
When will the next release be available? nightly builds?

### DISCUSSION THREAD

**Discussion by therealdreg** [bochs-emu/Bochs#61] [General]
We've made several changes to the repository & code.

When would it be a good idea to release a new version? It's been years since Bochs has released anything. 

It might also be interesting to always have a dev-testing version available. Surely, this way we would discover bugs earlier, and the project could progress a bit faster. 

What are your thoughts?

**Comment by stlintel** [comment]
Currently I am waiting for Ben Lunt to bring all his USB changes
I expect it won't happen too long.
Then we will need to regress everything seriously by running all OS images possible - it will take time for sure.
 About dev-testing-nightly version: not sure I know how to make github to build it and where it will stored.
Why enthusiasts cannot just compile it themselves from last snapshot ?

**Reply by therealdreg** [reply]
Compiling Bochs is not an easy task (a lot of options ..), and providing binary versions would be a much better idea. 

If a user wants to use the 'latest and greatest,' they should be able to do so with a single click. If they discover an issue, they can open a issue, and we can fix it before releasing the new version.

Frankly, expecting the average user to compile the latest version from the main branch may be somewhat naive. 

We need to assist our users in testing the product, don't you think?

If you like, I can create a PR with a new GitHub action to perform nightly builds, and we can always link to the latest version on the website. Does that sound appealing to you?

**Reply by stlintel** [reply]
Looks good.

Actually having compiler installed should be enough. Cygwin for Windows and gcc for Linux works with push button.
MSVCPP also should be push-button if we maintain MSVCPP project files (not sure we do now).
The build is tested by GitHub action on every push using the several configurations, include the 'release' configuration we build releases with.

We need Windows 64-bit and Linux build ?

**Reply by therealdreg** [reply]
I believe that having a dev-build for Linux x64 and another for Windows x64 is sufficient. 

But if we do this, what do you think is better? 

Having a GitHub action that we manually activate when the main branch is more 'stable' and we want to generate the dev-build? 

Or should it be done automatically every night?

**Reply by stlintel** [reply]
Let's start with manually activate. Then the repo is clean we could move to stable-weekly.

**Reply by therealdreg** [reply]
Okay, I think the same way. So, would you like me to create a github-action-PR to implement this?

**Reply by stlintel** [reply]
I am not a github expert, will learn from you on the way too :)

**Reply by therealdreg** [reply]
I don't have much expertise in this area, but I'll do the best I can!

**Reply by bachimusprime** [reply]
@stlintel I've tried to compile Bochs on Windows before, and it's a mess. For starters, the Makefile the configure script generates doesn't always work. Regenerating the configure script using autotools doesn't work either. Honestly, autotools is far too prone to breakage for my taste. Why not migrate to CMake or something? It's FAR more reliable.

**Reply by stlintel** [reply]
@fuel-pcbox: you might be didn't notice but there are nightly builds already. Every push to github trigger github action, you may enter Actions tab and download the latest Windows binary.

> For starters, the Makefile the configure script generates doesn't always work. 

It it doesn't better to open tracker item and discuss what doesn't work.
Clearly there is no need to regenerate configure scripts using autotools, it is good as is.
You may be don't have your cygwin/mingw installed correctly ...

**Reply by bachimusprime** [reply]
I do have my MSYS2 MinGW installed correctly, and I wanted to work on Bochs myself, so being able to compile from source is essential.

**Comment by Vort** [comment]
> It might also be interesting to always have a dev-testing version available.

I made PR with such change for Windows: #167.
It can be extended to other platforms and other compilers if needed.
Also it makes only single binary available for download, but it is possible to build whole installer in the same way.

Only downside is that download is allowed only for registered users of GitHub.
